Documentation update regarding newly supported device ODROID M1S

Jojo-A opened the pull request:
## Proposed change
Documentation update regarding newly supported device ODROID M1S.
The Documentation now refers to the ODROID M1S images, flashing gude has been created (similar to ODROID N2+)

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.


## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

Jojo-A committed:
Create flashing_m1s_otg

Initial commit with basic instructions regarding installing HAOS onto the ODROID M1S

Jojo-A committed:
Rename flashing_m1s_otg to flashing_m1s_otg.md

Jojo-A committed:
Update flashing_m1s_otg.md

Jojo-A committed:
Update flashing_m1s_otg.md

Jojo-A committed:
Update operating_system.md

Added first ODROID M1S references for OS selection

Jojo-A committed:
Update operating_system.md

Jojo-A committed:
Update os.markdown

added the link to the guide how to flash the ODROID M1S

Jojo-A committed:
Typo correction in flashing_m1s_otg.md

Jojo-A committed:
Update operating_system.md

List format correstion

darkxst commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,45 @@
+
+Home Assistant can be flashed to an ODROID-M1S by connecting the device directly to your computer via the USB-OTG connection on the front of the board. 
+Unlike other ODROID boards the M1S does not have a socket for an optionl eMMC module. It also does not have a separate flash chip that holds a dedicated bootloader.
+Instead the M1S has a build-in 64GB eMMC soldered directly on the board that holds a petitboot bootloader by default. This guide will show you how to flash HAOS to the build-in eMMC.
```
Please see the updates to the OS docs here, and reword in line with that and drop references to Petitboot: 
https://github.com/home-assistant/operating-system/blob/dev/Documentation/boards/hardkernel/odroid-m1s.md

It also includes a new recovery method that doesnt require using maskrom pin. Maybe you could add the Recovery steps as its own section?

Jojo-A replied:
Done. Updated the initial section and added a separate HK Recovery section

darkxst commented on source/_includes/installation/operating_system.md:
```diff
@@ -165,9 +165,11 @@ Use this method only if Method 1 does not work for you.
     {% endif %}
-3. Download and start <a href="https://www.balena.io/etcher" target="_blank">Balena Etcher</a>. You may need to run it with administrator privileges on Windows.
-4. Download the image to your computer.
+4. Download and start <a href="https://www.balena.io/etcher" target="_blank">Balena Etcher</a>. You may need to run it with administrator privileges on Windows.
```
why does the numbering change here?

Jojo-A replied:
Because of accident -_O
Fixed

darkxst commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,45 @@
+## Flashing an ODROID-M1S
+
+Home Assistant can be flashed to an ODROID-M1S by connecting the device directly to your computer via the USB-OTG connection on the front of the board. 
+Unlike other ODROID boards the M1S does not have a socket for an optionl eMMC module. It also does not have a separate flash chip that holds a dedicated bootloader.
```
typo, "optionl"

Jojo-A replied:
Done

darkxst commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,45 @@
+
+3. When the flash process is complete, disconnect the ODROID-M1S from your PC.
+   - Remove the power cable.
+   - Remove the micro-USB cabel
```
typo

Jojo-A replied:
Done

darkxst commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,45 @@
+      (balena Etcher will complain that something went wring during flashin. You can ignore this message)
+3. Plug-in that SD card to your ODROID-M1S and boot it.
+
+### Flashing Home Assistant
```
not sure you need to duplicate the steps from the main installation page?

Jojo-A replied:
Good point. True, there is nothing wrong in duplicating the description. But maybe just a link/reference to the flashing guide of the N2+ could be sufficient. I try to implement it to see how it looks like...

Jojo-A replied:
Okay, done. I hope the link to the N2+'s flashing guide works in the final version. Syntax should be fine from what I have read...

darkxst reviewed (changes requested):
@Jojo-A, Thanks for working on this, I've added a few comments.

You will also need to odroid M1S here
https://github.com/home-assistant/home-assistant.io/blob/1aaf0627932a1a3e121a7245e44bc0c4e1725b0b/_config.yml#L212

Jojo-A commented:
@darkxst Thank you very much for your feedback!
I will fix it of course. But please tell me: can I modify an issued PR later on? Or how does it work then? I will have to make new commits, sure. But how can/will these new commits be merged into this very PR? Sorry, still a git noob -_-

tl-sl commented:
Just push the new commits to same branch and the PR will pick up the new commits!

Jojo-A committed:
Update flashing_m1s_otg.md

- Typo corrections
- Section HK Recovery added

Jojo-A committed:
Update _config.yml

Added ODROID-M1S

Jojo-A committed:
Update operating_system.md

List correction

Jojo-A committed:
Update flashing_m1s_otg.md

- Removed duplicated instructions of how to flash the M1S
- Added a reference to the instructions of the N2+ because the are the same

darkxst commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+
+### Flashing Home Assistant M1S
+1. Download the latest stable version of Home Assistant OS for the [ODROID-M1S](https://github.com/home-assistant/operating-system/releases/download/{{site.data.version_data.hassos['odroid-m1s']}}/haos_odroid-m1s-{{site.data.version_data.hassos['odroid-m1s']}}.img.xz).
+2. Apart from the HAOS image to flash (M1S instead of N2+ version), you can follow the N2+ step-by-step flashing guide [HERE](/common-tasks/os/#flashing-home-assistant).
```
Just refer them back to the generic odroid intructions rather than N2 guide?

Jojo-A replied:
I think I don't agree here, because the reference at 2. is explicitly about the next steps how to flash the (internal) eMMC. Sure, a reference to the generic ODROID instructions are not entirely "wrong", but I think the reference to the really next steps makes more sense.

darkxst commented on source/_includes/installation/operating_system.md:
```diff
@@ -183,6 +185,8 @@ Use this method only if Method 1 does not work for you.
     {% if variant.key == "odroid-n2" %}
     [Guide: Flashing Odroid-N2 using OTG-USB](/hassio/flashing_n2_otg/)
+    {% elsif variant.key == "odroid-m1s" %}
+    [Guide: Flashing Odroid-M1S using OTG-USB](/hassio/flashing_m1s_otg/)
```
Guide: Boot Odroid-M1S into UMS

Jojo-A replied:
Could you explain why you might prefer that title? I ask because from my point of view the procedure is the same as for the N2+. So from this perspective it is just consistent to also have the same title.
Or the other way: wouldn't your title proposition also apply to the N2+? If yes: why not keeping it the same for the sake of consistency?

tl-sl replied:
I suppose that doesnt apply if your not referring back to these steps after.. Keep it the same as N2 if you prefer

Jojo-A replied:
> I suppose that doesnt apply if your not referring back to these steps after.. Keep it the same as N2 if you prefer

What do you mean by that? Do you agree to @darkxst to change the guides title? Or do you think that keeping the name identical to the N2 stuff is okay because the following steps / the content is basically the same?

darkxst replied:
sorry that was my comment also, but from wrong account! Keep it the same as N2 if you prefer. However I guess keep the duplicated steps in this case and not rename this.

Jojo-A replied:
Alright, than I would leave it as it currently is. Maybe it is a matter of personal opinion if one prefers cross references over duplicated content or vice versa. If you are fie with the current state I mark this conversation as resolved.

darkxst replied:
Ok, LGTM.

You could probably add the board to this list also:
https://github.com/home-assistant/home-assistant.io/blob/current/source/_includes/installation/operating_system.md#suggested-hardware

Jojo-A replied:
> You could probably add the board to this list also: https://github.com/home-assistant/home-assistant.io/blob/current/source/_includes/installation/operating_system.md#suggested-hardware

Well, yes and no. I was aware of that page and thought about that, too. But the section is written from the perspective of the HA administration with the wording "we recommend". This is not the position I am in. Additionally a device support will not automatically mean a recommendation (like the C2, that is supported but not explicitly recommended).
So I think to list the device under "recommendations" it will require either more user experiences or objective benchmarks as a HA device to officially recommend it.
Apart from that I would have to create affiliate links, that I just can't. AND: ameridroid seems not to have the M1S in their catalog at all (at the time of writing).

To sum it up:
- mention in the download section: yes (done)
- mention in recommendations: not yet possible
- add to list of affiliate links: not possible (for me) due to not available in online shop

What do you think?

darkxst replied:
Yeh they do have it:
https://ameridroid.com/products/odroid-m1s

> Additionally a device support will not automatically mean a recommendation (like the C2, that is supported but not explicitly recommended).

Right if it was a completely new board/soc could be different, but since its only slightly modified from M1, support should be as good as that from day one. C2 is not a good comparison, it is obsolete and discontinued! Anyway see what @sairon thinks for this.

Jojo-A replied:
> Yeh they do have it: https://ameridroid.com/products/odroid-m1s
> 
Oh, my bad! I went through the SBC sub menu of ameridroid and did not find it. Maybe I should have tried the text search...

Jojo-A replied:
Done

darkxst commented on source/_includes/installation/operating_system.md:
```diff
@@ -212,7 +216,7 @@ Use this method only if Method 1 does not work for you.
    - If the operation fails, decompress the .xz file and try again.
 ![Screenshot of the Etcher software showing the Flash button highlighted.](/images/installation/etcher5.png)
-1.  When Balena Etcher has finished writing the image, you will see a confirmation.
+   1.  When Balena Etcher has finished writing the image, you will see a confirmation.
```
this looks odd with a number, should it be a bullet point?

Jojo-A replied:
Good point. I stumbled across that as well because that weird numbering is there in the current guide. So I thought there might have been a reason for that. I personally am fine with changing that.

sairon commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+
+
+#### _HK Recovery_
+If you want to restore your M1S back into Hardkernels initial state, you will have to restore the HK's bootloader.
```
```suggestion
If you want to restore your M1S back into Hardkernel's initial state, you will have to restore the HK's bootloader.
```

sairon commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+
+Home Assistant can be flashed to an ODROID-M1S by connecting the device directly to your computer via the USB-OTG connection on the front of the board. 
+Unlike other ODROID boards the M1S does not have a socket for an optional eMMC module. It also does not have a separate flash chip that holds a dedicated bootloader.
+Instead the M1S has a build-in 64GB eMMC soldered directly on the board that holds a petitboot bootloader by default. This guide will show you how to flash HAOS to the build-in eMMC.
```
```suggestion
Instead the M1S has a build-in 64GB eMMC soldered directly on the board that holds a bootloader by default. This guide will show you how to flash HAOS to the build-in eMMC.
```

There's no petitboot - basically it's ARM firmware, followed by U-Boot SPL and U-Boot proper, but we can spare the reader of the gory details, just drop the "petitboot".

Jojo-A replied:
Thank you for pointing that out. @darkxst already suggested to remove explicit "petitboot" references. That one here just slipped through...

sairon reviewed (commented):
One nit about the bootloader below. Also we should use consistent naming of the board in the text (the correct one is `ODROID-M1S` - all caps with a dash). That's a broader issue that should be addressed in another PR, but maybe we could use consistent name for M1S for new occurrences at least.

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: Jan Čermák <sairon@users.noreply.github.com>

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: Jan Čermák <sairon@users.noreply.github.com>

Jojo-A committed:
Update operating_system.md

- minor correction in list formatting
- unified naming of ODROID devices

Jojo-A commented:
> One nit about the bootloader below. Also we should use consistent naming of the board in the text (the correct one is `ODROID-M1S` - all caps with a dash). That's a broader issue that should be addressed in another PR, but maybe we could use consistent name for M1S for new occurrences at least.

Thats definitely true! I already corrected the inconsistent names I was recently stumbling across. But for further corrections I think this is not critical and could also be merged into the current branch.

darkxst commented:
@Jojo-A This probably should be against the `current` branch, as OS 12.0 is releasing today.

sairon reviewed (dismissed):
The content looks good to me, I'll hand it over to @c0ffeeca7 now to handle the formal stuff.

c0ffeeca7 commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+## Flashing an ODROID-M1S
+
+Home Assistant can be flashed to an ODROID-M1S by connecting the device directly to your computer via the USB-OTG connection on the front of the board. 
+Unlike other ODROID boards the M1S does not have a socket for an optional eMMC module. It also does not have a separate flash chip that holds a dedicated bootloader.
```
```suggestion
Unlike other ODROID boards, the M1S does not have a socket for an optional  <abbr title="embedded MultiMediaCard">eMMC</abbr> module. It also does not have a separate flash chip that holds a dedicated bootloader.
```

c0ffeeca7 commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+
+Home Assistant can be flashed to an ODROID-M1S by connecting the device directly to your computer via the USB-OTG connection on the front of the board. 
+Unlike other ODROID boards the M1S does not have a socket for an optional eMMC module. It also does not have a separate flash chip that holds a dedicated bootloader.
+Instead the M1S has a build-in 64GB eMMC soldered directly on the board that holds a bootloader by default. This guide will show you how to flash HAOS to the build-in eMMC.
```
```suggestion
Instead, the M1S has a build-in 64GB <abbr title="embedded MultiMediaCard">eMMC</abbr> soldered directly on the board that holds a bootloader by default. This guide will show you how to install the Home Assistant Operating System to the built-in <abbr title="embedded MultiMediaCard">eMMC</abbr>.
```

c0ffeeca7 commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+Unlike other ODROID boards the M1S does not have a socket for an optional eMMC module. It also does not have a separate flash chip that holds a dedicated bootloader.
+Instead the M1S has a build-in 64GB eMMC soldered directly on the board that holds a bootloader by default. This guide will show you how to flash HAOS to the build-in eMMC.
+
+<ins>**Warning:</ins> Installing HAOS replaces the firmware and SPL on the eMMC with the mainline version provided by HAOS. As a result, it is not possible to use the SD card with the EMMC2UMS image anymore, because the mainline SPL is not compatible with U-Boot in the EMMC2UMS image at this time (February 2024). This does not pose any problem for standard use, just makes it more complicated in c ...
```
```suggestion
<ins>**Warning:</ins> Installing Home Assistant OS replaces the firmware and <abbr title="secondary program loader">SPL</abbr> on the <abbr title="embedded MultiMediaCard">eMMC</abbr> with the mainline version provided by the Home Assistant OS. As a result, it is not possible to use the SD card with the EMMC2UMS image anymore, because the mainline <abbr title="secondary program loader">SPL</abbr> is not compatible with U-Boot in the EMMC2UMS image at this time (February 2024). This does not pose any problem for standard use, just makes it more complicated in case you want to return to the Hardkernel-provided OS (see HK Recovery).**
```

c0ffeeca7 commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+
+### What you will need
+
+To flash your eMMC using OTG-USB, you will need the following items:
```
```suggestion
To flash your <abbr title="embedded MultiMediaCard">eMMC</abbr> using <abbr title="USB-On-The-Go">USB-OTG</abbr>, you will need the following items:
```

c0ffeeca7 commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+- USB 2.0 to micro-USB cable
+- the special Hardkernel eMMC-to-USB-mass-storage image
+
+### Boot into mass-storage mode
```
```suggestion
### Boot into mass-storage mode

```

c0ffeeca7 commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+### Boot into mass-storage mode
+(These steps are identical to the official [Hardkernel wiki](https://wiki.odroid.com/odroid-m1s/getting_started/os_installation_guide?redirect=1#install_over_usb_from_pc) page.)
+
+1. Download [ODROID-M1S_EMMC2UMS.img](https://dn.odroid.com/RK3566/ODROID-M1S/Installer/ODROID-M1S_EMMC2UMS.img)
```
```suggestion
1. Download [ODROID-M1S_EMMC2UMS.img](https://dn.odroid.com/RK3566/ODROID-M1S/Installer/ODROID-M1S_EMMC2UMS.img).
```

c0ffeeca7 commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+
+1. Download [ODROID-M1S_EMMC2UMS.img](https://dn.odroid.com/RK3566/ODROID-M1S/Installer/ODROID-M1S_EMMC2UMS.img)
+2. Use [balena Etcher](https://www.balena.io/etcher/) or another tool to flash the UMS utility onto an SD card.
+   - use **Flash from file**. **Flash from URL** does not work on all systems.
```
```suggestion
   - Use **Flash from file**. **Flash from URL** does not work on all systems.
```

c0ffeeca7 commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+1. Download [ODROID-M1S_EMMC2UMS.img](https://dn.odroid.com/RK3566/ODROID-M1S/Installer/ODROID-M1S_EMMC2UMS.img)
+2. Use [balena Etcher](https://www.balena.io/etcher/) or another tool to flash the UMS utility onto an SD card.
+   - use **Flash from file**. **Flash from URL** does not work on all systems.
+      (balena Etcher will complain that something went wring during flashin. You can ignore this message)
```
```suggestion
      (balena Etcher will complain that something went wrong during flashing. You can ignore this message)
```

c0ffeeca7 commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+      (balena Etcher will complain that something went wring during flashin. You can ignore this message)
+3. Plug-in that SD card to your ODROID-M1S and boot it.
+
+### Flashing Home Assistant M1S
```
```suggestion
### Flashing Home Assistant M1S

```

c0ffeeca7 commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+2. Apart from the HAOS image to flash (M1S instead of N2+ version), you can follow the N2+ step-by-step flashing guide [HERE](/common-tasks/os/#flashing-home-assistant).
+
+
+#### _HK Recovery_
```
```suggestion
#### _HK Recovery_

```

c0ffeeca7 commented on source/_includes/common-tasks/flashing_m1s_otg.md:
```diff
@@ -0,0 +1,41 @@
+
+#### _HK Recovery_
+If you want to restore your M1S back into Hardkernel's initial state, you will have to restore the HK's bootloader.
+A reliable way of reflashing the eMMC in this case is to use HAOS booted from an SD card. To do that, insert the SD card with HAOS to the micro SD slot and plug the board in. Once the device boots to the HA CLI, enter `login` to enter the root shell and use `curl` to download an image and `dd` it to the eMMC block device:
```
```suggestion
A reliable way of reflashing the eMMC in this case is to use Home Assistant OS booted from an SD card. To do that, insert the SD card with Home Assistant OS to the micro SD slot and plug the board in. Once the device boots to the Home Assistant CLI (command line), enter `login` to enter the root shell and use `curl` to download an image and `dd` it to the eMMC block device:
```

c0ffeeca7 commented on source/_includes/common-tasks/flashing_m1s_otg.md:
````diff
@@ -0,0 +1,41 @@
+curl https://dn.odroid.com/RK3566/ODROID-M1S/Installer/ODROID-M1S_EMMC2UMS.img | dd of=/dev/mmcblk0
+```
+
+This way the device will start in the UMS mode on the next boot with the SD card removed. Alternatively you can use the [Hardkernel installer image](https://wiki.odroid.com/odroid-m1s/getting_started/os_installation_guide#user_installer) directly instead of the EMMC2UMS image.
````
```suggestion
This way, the device will start in the UMS mode on the next boot with the SD card removed. Alternatively, you can use the [Hardkernel installer image](https://wiki.odroid.com/odroid-m1s/getting_started/os_installation_guide#user_installer) directly instead of the EMMC2UMS image.
```

agners commented on source/_includes/common-tasks/flashing_m1s_otg.md:
````diff
@@ -0,0 +1,41 @@
+curl https://dn.odroid.com/RK3566/ODROID-M1S/Installer/ODROID-M1S_EMMC2UMS.img | dd of=/dev/mmcblk0
+```
+
+This way the device will start in the UMS mode on the next boot with the SD card removed. Alternatively you can use the [Hardkernel installer image](https://wiki.odroid.com/odroid-m1s/getting_started/os_installation_guide#user_installer) directly instead of the EMMC2UMS image.
````
I don't think this should be in the user documentation. While convenient during development and for those know what they do, for lots of users this is probably confusing. We have it in the more power-user/developer oriented documentation in the OS repo (https://github.com/home-assistant/operating-system/blob/e60a2e4b68c061b920347b9394e5b95696d306cd/Documentation/boards/hardkernel/odroid-m1s.md#emmc).

Jojo-A replied:
Thank you for reviewing that.


> I don't think this should be in the user documentation. While convenient during development and for those know what they do, for lots of users this is probably confusing. We have it in the more power-user/developer oriented documentation in the OS repo (https://github.com/home-assistant/operating-system/blob/e60a2e4b68c061b920347b9394e5b95696d306cd/Documentation/boards/hardkernel/odroid-m1s.md#emmc).

I personally am fine with keeping it there, because even as a less experienced user it would make me feel just comfortable if I directly see, that there is a "way back" from what I have been warned before. Or the other way: if I get warned about something and I don't see how/if I could revert that, this would make me kind of nervous.
If course I can remove it again and maybe just refer to the power-user guide if you or the majority asks me to do so. But my guts are fine with keeping it there ;) .

agners replied:
Sorry, my comment was a bit rushed and unclear.

> [...] it would make me feel just comfortable if I directly see, that there is a "way back" from what I have been warned before.

I agree with that.

However, the method described here is somewhat advanced. Folks will type this command into the web terminal, where access to  `/dev/mmcblk0` is not possible. The URL might become invalid. Every documentation needs to be maintained, if users are unclear issues being handled etc.  Sometimes more is more, but sometimes less is more :sweat_smile: 

There is a more generic way of recovering, which doesn't even need terminal access. and we should advice to use this by default. So I'd suggest to just document the Hardkernel installer image method, and link to the external wiki in this user documentation.

darkxst replied:
> There is a more generic way of recovering, which doesn't even need terminal access. and we should advice to use this by default. So I'd suggest to just document the Hardkernel installer image method, and link to the external wiki in this user documentation.

The Hardkernel images do not boot with mainline u-boot installed on eMMC though. That was the entire reason for the workaround in the first place. The only other option is shorting out maskROM pads to boot hardkernel images.

agners replied:
@darkxst so then this sentence is wrong?

> Alternatively you can use the [Hardkernel installer image](https://wiki.odroid.com/odroid-m1s/getting_started/os_installation_guide#user_installer) directly instead of the EMMC2UMS image.

agners replied:
Ah I understand, it is probably meant to be flashed using dd directly, similarly to ODROID-M1S_EMMC2UMS.img.

A few problems though: The command would be different as that image is xz compresseed. And this is definitely is not safe/guarnateed to work, as the image is larger and would overwrite the squashfs rootfs partition of HAOS. :cry:

agners replied:
Oh, nevermind, the commands are meant to be run from a HAOS running on a SD card. Yeah ok, then both methods should work and are safe.

The Hardkernel installer image can't be installed directly though because HAOS is missing unxz :cry: 

Also, I'd say that the command should work safely when booted from eMMC. The image is only 3MB, so this does not alter any of the mounted partitions. Having two HAOS installation on separate partition usually ends up in a bit chaotic situations since we rely on static partition UUID to find rootfs partitions etc.

I'll make a suggestion against the updated PR.

darkxst replied:
> Ah I understand, it is probably meant to be flashed using dd directly, similarly to ODROID-M1S_EMMC2UMS.img.

yes that is it, sairon added that comment to the docs, but guess he didnt test it

darkxst replied:
> The only other option is shorting out maskROM pads to boot hardkernel images.

One other option is to make our own UMS tool with mainline uboot. It is just uboot with UMS enabled in that Hardkernel image.

Jojo-A replied:
I added the suggestion of @agners (see https://github.com/home-assistant/home-assistant.io/pull/31540/commits/86671fc6a59eac5d2e8fde9e6a9383b85df4fb25).
Can I clase this one as resolved?

sairon replied:
@darkxst 
> The only other option is shorting out maskROM pads to boot hardkernel images.

But IIUC that should get you to the Rockchip download mode, which is even more involved procedure :grimacing:

> yes that is it, sairon added that comment to the docs, but guess he didnt test it

Yes, indeed I did not :sweat_smile: A simple piping through `xz -dc` should solve it, I guess I should be more explicit in the OS docs.

darkxst replied:
> But IIUC that should get you to the Rockchip download mode, which is even more involved procedure 😬

Only if it can't boot from SD card, will it fallback to RockUSB.

sairon replied:
Ah, you're right. I got the impression it doesn't boot from the SD from [this post](https://forum.odroid.com/viewtopic.php?p=377662&sid=3fc9d7429603736e8dd7c2ca4c1ff904#p377662) but I just tested it behaves the way you say :+1:

c0ffeeca7 reviewed (commented):
Some style tweaks

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

c0ffeeca7 commented:
Should we add the boards also to the Suggested hardware section?

- [ODROID M1S: 4 GB RAM / 64 GB eMMC](https://ameridroid.com/products/odroid-m1s?variant=47425396474135?ref=eeb6nfw07e)
- [ODROID M1S: 8 GB RAM / 64 GB eMMC](https://ameridroid.com/products/odroid-m1s?variant=47425396506903?ref=eeb6nfw07e)

Jojo-A commented:
> Should we add the boards also to the Suggested hardware section?
> 
>     * [ODROID M1S: 4 GB RAM / 64 GB eMMC](https://ameridroid.com/products/odroid-m1s?variant=47425396474135?ref=eeb6nfw07e)
> 
>     * [ODROID M1S: 8 GB RAM / 64 GB eMMC](https://ameridroid.com/products/odroid-m1s?variant=47425396506903?ref=eeb6nfw07e)

Well, I can only speak for myself. So I personally am quite satisfied with the device and the usecase as a HA server. But I also think, that performance on XU4 had still been a little bit better (but I switched to M1S because of 64bit architecture and HK has marked the XU4 as discontinued).
But @darkxst is right when he says that it is a very similar board to the M1 (which is suggested by the documentation). So if the M1 is listed, the M1S should be listed as well...

Jojo-A committed:
Update operating_system.md

- added ODROID-M1S to suggested hardware
- corrections in ODROID device naming

c0ffeeca7 reviewed (commented):
:heart: Thank you!

Jojo-A commented:
> > Should we add the boards also to the Suggested hardware section?
> > ```
> > * [ODROID M1S: 4 GB RAM / 64 GB eMMC](https://ameridroid.com/products/odroid-m1s?variant=47425396474135?ref=eeb6nfw07e)
> > 
> > * [ODROID M1S: 8 GB RAM / 64 GB eMMC](https://ameridroid.com/products/odroid-m1s?variant=47425396506903?ref=eeb6nfw07e)
> > ```
> 
> Well, I can only speak for myself. So I personally am quite satisfied with the device and the usecase as a HA server. But I also think, that performance on XU4 had still been a little bit better (but I switched to M1S because of 64bit architecture and HK has marked the XU4 as discontinued). But @darkxst is right when he says that it is a very similar board to the M1 (which is suggested by the documentation). So if the M1 is listed, the M1S should be listed as well...

Done

Jojo-A commented:
Shall I rebase the PR because M1S is supported since last release?

agners commented:
@Jojo-A the branching of home-assistant.io tracks Core. It doesn't really work with HAOS related content :cry: 

So yeah, best would be if you can rebase this on current and target against current. It seems to be possible to change the target branch.

agners commented on source/_includes/common-tasks/flashing_m1s_otg.md:
````diff
@@ -0,0 +1,44 @@
+#### _HK Recovery_
+
+If you want to restore your M1S back into Hardkernel's initial state, you will have to restore the HK's bootloader.
+A reliable way of reflashing the eMMC in this case is to use Home Assistant OS booted from an SD card. To do that, insert the SD card with Home Assistant OS to the micro SD slot and plug the board in. Once the device boots to the Home Assistant CLI (command line), enter `login` to enter the root shell and use `curl` to download an image and `dd` it to the eMMC block device:
+
+```sh
+curl https://dn.odroid.com/RK3566/ODROID-M1S/Installer/ODROID-M1S_EMMC2UMS.img | dd of=/dev/mmcblk0
+```
+
+This way, the device will start in the UMS mode on the next boot with the SD card removed. Alternatively, you can use the [Hardkernel installer image](https://wiki.odroid.com/odroid-m1s/getting_started/os_installation_guide#user_installer) directly instead of the EMMC2UMS image.
````
```suggestion
A reliable way of reflashing the eMMC with an operating system of youur choice is to use Home Assistant OS to flash the EMMC2UMS image which turns the ODROID-M1S into USB Mass Storage device. Once you have flashed the EMMC2UMS image, you can flash any OS again. You will need a micro USB cable to connect ODROID-M1S to PC.

Note: This commands will render your current Home Assistant OS installation unbootable!

Use the local terminal (HDMI/keyboard) to access the system console. On the Home Assistant CLI (command line), enter `login` to enter the root shell and use `curl` to download an image and `dd` it to the eMMC block device:

```sh
curl https://dn.odroid.com/RK3566/ODROID-M1S/Installer/ODROID-M1S_EMMC2UMS.img | dd of=/dev/mmcblk0
```

This way, the device will start in the UMS mode on the next boot with the SD card removed. Follow the [Install over USB from PC](https://wiki.odroid.com/odroid-m1s/getting_started/os_installation_guide#install_over_usb_from_pc) to install a different operating system.
```

Jojo-A replied:
corrected a typo and added the last part that you seem to have lost (or just weird formatted in the suggestion post ;) )

agners replied:
@Jojo-A uh whoops, I see thanks.

However, it seems the link got lost. Can you make sure that we link to https://wiki.odroid.com/odroid-m1s/getting_started/os_installation_guide#install_over_usb_from_pc in that last sentence?

Jojo-A replied:
Oh, whoops!
Done :) .

Jojo-A committed:
Update source/_includes/common-tasks/flashing_m1s_otg.md

Co-authored-by: Stefan Agner <stefan@agner.ch>

Jojo-A committed:
Update flashing_m1s_otg.md

- typo correction
- added missing description

Jojo-A committed:
Update flashing_m1s_otg.md

link added

Jojo-A committed:
Update flashing_m1s_otg.md

- anchor link fixed

Jojo-A commented:
> @Jojo-A the branching of home-assistant.io tracks Core. It doesn't really work with HAOS related content 😢
> 
> So yeah, best would be if you can rebase this on current and target against current. It seems to be possible to change the target branch.

I think I need a hint how to rebase it. I have the option to "Update branch" and "Update with rebase". But the latter one does not change the target branch, it just rebases the PR to the most recent version of the current branch. So how to proceed now?

sairon commented:
@Jojo-A Sorry, I wanted to help you out there, but actually ended up screwing things up :facepalm: I rebased your changes onto the `current` branch and wanted it to force-push to your PR branch, but I pushed the current `next` to your fork instead of the rebased commits. And based on that, Github assumed the PR is closed with no changes and doesn't let me reopen the PR or push to your fork anymore :see_no_evil: 

I pushed the rebased changes to [Jojo-A-ODROID-M1S](https://github.com/home-assistant/home-assistant.io/pull/new/Jojo-A-ODROID-M1S) branch for now. Either you can sync that with your fork and open a new PR against the `current` branch, or I _think_ that  you could be able to re-sync your `next` branch in your fork and the PR could be re-opened then if you modify any arbitrary file in that branch (because currently GH thinks there's nothing new). Then I could (hopefully) force-push the correct patches after this PR is open.

Again, I'm sorry for the mess.

Jojo-A commented:
> @Jojo-A Sorry, I wanted to help you out there, but actually ended up screwing things up 🤦 I rebased your changes onto the `current` branch and wanted it to force-push to your PR branch, but I pushed an old revision of `next` to your fork instead of the rebased commits. And based on that, Github assumed the PR is closed with no changes and doesn't let me reopen the PR or push to your fork anymore 🙈
> 
> I pushed the rebased changes to [Jojo-A-ODROID-M1S](https://github.com/home-assistant/home-assistant.io/pull/new/Jojo-A-ODROID-M1S) branch for now. Either you can sync that with your fork and open a new PR against the `current` branch, or I _think_ that you could be able to re-sync your `next` branch in your fork and the PR could be re-opened then if you modify any arbitrary file in that branch (because currently GH thinks there's nothing new). Then I could (hopefully) force-push the correct patches after this PR is open.
> 
> Again, I'm sorry for the mess.

Haha, now imagine how all this sound to me as a github noob :sweat_smile: !

Alright, if I follow your link to the  [Jojo-A-ODROID-M1S](https://github.com/home-assistant/home-assistant.io/pull/new/Jojo-A-ODROID-M1S) branch I get directly asked if I want to open a PR from that branch, where current is set as default target branch. Shall I just follow these steps? I ask because it seems quite straight forward - which makes me a little bit skeptic if things that I don't understand seem too easy :sweat_smile: ...

sairon commented:
> Haha, now imagine how all this sound to me as a github noob 😅 !

Yes, I can relate. And even when you think that you know Git, there are still ways it can screw you over :sweat_smile: 

> Alright, if I follow your link to the [Jojo-A-ODROID-M1S](https://github.com/home-assistant/home-assistant.io/pull/new/Jojo-A-ODROID-M1S) branch I get directly asked if I want to open a PR from that branch, where current is set as default target branch.

Yes, that might work - that way the PR will be opened on your behalf, but you won't be able to modify anything in that branch anymore.

So I would still like to try if it's possible to do that the way I suggested above. I.e. sync your forked `next` branch using the button in [your fork](https://github.com/Jojo-A/home-assistant.io/tree/next):
![image](https://github.com/home-assistant/home-assistant.io/assets/211416/0a1cccff-bf05-4352-9a27-fbdb14d3992d)

Then edit any file (shouldn't matter which one), and that point it _should be_ possible to re-open this PR, and with allowed edits from maintainers, let me force-push the correct commits.

Jojo-A commented:
> Then edit any file (shouldn't matter which one), and that point it _should be_ possible to re-open this PR, and with allowed edits from maintainers, let me force-push the correct commits.

Okay, did that. But until now I don't see an option to re-open this PR again.
I only have the option to just open a new PR:
![Bildschirmfoto vom 2024-02-29 14-06-57](https://github.com/home-assistant/home-assistant.io/assets/33325587/30c1f603-6280-447b-a487-8aae3bcebb2f)

and looking at the commits recently done:
![Bildschirmfoto vom 2024-02-29 14-07-53](https://github.com/home-assistant/home-assistant.io/assets/33325587/556740a9-aac2-4da5-a5fa-ea32116265ab)

c0ffeeca7 reviewed (approved):
Thank you, @Jojo-A, @darkxst, @sairon, @agners   :+1:

Jojo-A commented:
> Thank you, @Jojo-A 👍

Thank YOU guys! :blue_heart:
